File: ai_tnhn/ai-api-tnhn/router/emergency_construction.go
package router

import (
	"ai-api-tnhn/handler"
	"ai-api-tnhn/router/middleware"

	"github.com/gin-gonic/gin"
)

func (h HandlerFuncs) EmergencyConstructionRoutes(rg *gin.RouterGroup, mid middleware.Middleware, handler *handler.EmergencyConstructionHandler) {
	group := rg.Group("/emergency-constructions")
	group.Use(mid.MidBasicType())
	{
		group.GET("", handler.List)
		group.GET("/history", handler.ListHistory)
		group.POST("", handler.Create)
		group.GET("/:id", handler.GetByID)
		group.PUT("/:id", handler.Update)
		group.DELETE("/:id", handler.Delete)
		group.GET("/:id/history", handler.GetHistory)
		group.GET("/progress/:id", handler.GetProgressByID)
		group.POST("/progress", handler.ReportProgress)
		group.PUT("/progress/:id", handler.UpdateProgress)
		group.GET("/:id/progress", handler.GetProgressHistory)
		group.GET("/export", handler.ExportExcel)
	}
}

File: ai_tnhn/ai-frontend-tnhn/src/api/emergencyConstruction.js
import axiosClient from './axiosClient';

const emergencyConstructionApi = {
    getAll: (params) => {
        return axiosClient.get('/admin/emergency-constructions', { params });
    },
    getById: (id) => {
        return axiosClient.get(`/admin/emergency-constructions/${id}`);
    },
    create: (data) => {
        return axiosClient.post('/admin/emergency-constructions', data);
    },
    update: (id, data) => {
        return axiosClient.put(`/admin/emergency-constructions/${id}`, data);
    },
    delete: (id) => {
        return axiosClient.delete(`/admin/emergency-constructions/${id}`);
    },
    getHistory: (id) => {
        return axiosClient.get(`/admin/emergency-constructions/${id}/history`);
    },
    getGlobalHistory: (params) => {
        return axiosClient.get('/admin/emergency-constructions/history', { params });
    },
    createProgress: (data) => {
        if (data instanceof FormData) {
            return axiosClient.post('/admin/emergency-constructions/progress', data, {
                headers: { 'Content-Type': 'multipart/form-data' }
            });
        }
        return axiosClient.post('/admin/emergency-constructions/progress', data);
    },
    updateProgress: (id, data) => {
        if (data instanceof FormData) {
            return axiosClient.put(`/admin/emergency-constructions/progress/${id}`, data, {
                headers: { 'Content-Type': 'multipart/form-data' }
            });
        }
        return axiosClient.put(`/admin/emergency-constructions/progress/${id}`, data);
    },
    getProgressById: (id) => {
        return axiosClient.get(`/admin/emergency-constructions/progress/` + id);
    },
    getProgressHistory: (id) => {
        return axiosClient.get(`/admin/emergency-constructions/${id}/progress`);
    }
};

export default emergencyConstructionApi;
